src/traces/scatter/index.js

```diff
@@ -855,3 +857,53 @@ scatter.hoverPoints = function(pointData, xval, yval, hovermode) {
 
     return [pointData];
 };
+
+var DESELECTDIM = 0.2;
+
+scatter.selectPoints = function(searchInfo, polygon) {
+    var cd = searchInfo.cd,
+        xa = searchInfo.xaxis,
+        ya = searchInfo.yaxis,
+        selection = [],
+        curveNumber = cd[0].trace.index,
+        marker = cd[0].trace.marker,
+        i,
+        di,
+        x,
+        y;
+
+    if(!marker) return; // TODO: include text and/or lines?
+
+    var opacity = Array.isArray(marker.opacity) ? 1 : marker.opacity;
+
+    if(polygon === false) { // clear selection
+        for(i = 0; i < cd.length; i++) cd[i].dim = 0;
+    }
+    else {
+        for(i = 0; i < cd.length; i++) {
+            di = cd[i];
+            x = xa.c2p(di.x);
+            y = ya.c2p(di.y);
+            if(polygon.contains([x, y])) {
+                selection.push({
+                    curveNumber: curveNumber,
+                    pointNumber: i,
+                    x: di.x,
+                    y: di.y
+                });
```

Comment:
@alexcjohnson you dropped the other parts of https://github.com/plotly/plotly.js/blob/master/src/plots/cartesian/graph_interact.js#L549-L556 for speed I imagine?